train.py

```diff
@@ -40,6 +41,10 @@
                      'Number of training steps to take. If not set, iterates '
                      'once over training data.')
 
+flags.DEFINE_integer('num_examples', None,
+                     'Total number of examples passed. Used to calculate '
```

Comment:
nit: i'm really scratching my head.  consider "This is only used if `steps_to_train` is not set."

(do we need a validator for at least one of them being set?)

Reply:
done